hugr-core/src/macros.rs

```diff
@@ -57,7 +57,7 @@ pub(crate) use impl_box_clone;
 macro_rules! type_row {
     () => {
         {
-            $crate::types::TypeRow::new()
+            $crate::types::TypeRow::<false>::new()
```

Comment:
that `type_row!` can only be used for `TypeRow<false>` is unfortunate but unavoidable (`static` stuff needs an explicit type). So this makes the 0-ary case consistent with the non-empty cases.

We could add a `type_rowv!` or similar for `TypeRow<true>`, I suppose....